diff --git a/dist/cubism4.es.js b/dist/cubism4.es.js
index eb264b4709d4aa2d8b05debc823fbc9bf0aaa51b..bf99ae69b9b9c50dce121c63c9578232e9d1693c 100644
--- a/dist/cubism4.es.js
+++ b/dist/cubism4.es.js
@@ -2284,7 +2284,7 @@ class CubismModel {
    * @return Drawableの描画順リスト
    */
   getDrawableRenderOrders() {
-    return this._model.drawables.renderOrders;
+    return this._model.drawables.renderOrders ?? this._model.drawables.drawOrders;
   }
   /**
    * @deprecated
@@ -7380,7 +7380,7 @@ class CubismRenderer_WebGL extends CubismRenderer {
   release() {
     var _a, _b, _c;
     const self2 = this;
-    this._clippingManager.release();
+    this._clippingManager?.release();
     self2._clippingManager = void 0;
     (_a = this.gl) == null ? void 0 : _a.deleteBuffer(this._bufferData.vertex);
     this._bufferData.vertex = null;
@@ -7409,9 +7409,9 @@ class CubismRenderer_WebGL extends CubismRenderer {
     const drawableCount = this.getModel().getDrawableCount();
     const renderOrder = this.getModel().getDrawableRenderOrders();
     for (let i = 0; i < drawableCount; ++i) {
-      const order = renderOrder[i];
-      this._sortedDrawableIndexList[order] = i;
+      this._sortedDrawableIndexList[i] = i;
     }
+    this._sortedDrawableIndexList.sort((a, b) => renderOrder[a] - renderOrder[b]);
     for (let i = 0; i < drawableCount; ++i) {
       const drawableIndex = this._sortedDrawableIndexList[i];
       if (!this.getModel().getDrawableDynamicFlagIsVisible(drawableIndex)) {
diff --git a/dist/cubism4.js b/dist/cubism4.js
index 70e107d6cc407a02fdebef231ee855626f43cb49..e1de0e040000676e8fb10660164fe790d29e36a8 100644
--- a/dist/cubism4.js
+++ b/dist/cubism4.js
@@ -2286,7 +2286,7 @@ var __async = (__this, __arguments, generator) => {
      * @return Drawableの描画順リスト
      */
     getDrawableRenderOrders() {
-      return this._model.drawables.renderOrders;
+      return this._model.drawables.renderOrders ?? this._model.drawables.drawOrders;
     }
     /**
      * @deprecated
@@ -7382,7 +7382,7 @@ var __async = (__this, __arguments, generator) => {
     release() {
       var _a, _b, _c;
       const self2 = this;
-      this._clippingManager.release();
+      this._clippingManager?.release();
       self2._clippingManager = void 0;
       (_a = this.gl) == null ? void 0 : _a.deleteBuffer(this._bufferData.vertex);
       this._bufferData.vertex = null;
@@ -7411,9 +7411,9 @@ var __async = (__this, __arguments, generator) => {
       const drawableCount = this.getModel().getDrawableCount();
       const renderOrder = this.getModel().getDrawableRenderOrders();
       for (let i = 0; i < drawableCount; ++i) {
-        const order = renderOrder[i];
-        this._sortedDrawableIndexList[order] = i;
+        this._sortedDrawableIndexList[i] = i;
       }
+      this._sortedDrawableIndexList.sort((a, b) => renderOrder[a] - renderOrder[b]);
       for (let i = 0; i < drawableCount; ++i) {
         const drawableIndex = this._sortedDrawableIndexList[i];
         if (!this.getModel().getDrawableDynamicFlagIsVisible(drawableIndex)) {
diff --git a/dist/index.es.js b/dist/index.es.js
index 7ca11af82cdde3564816a6e0b839d7dbcaa6b11f..9900ea893d3645215d8087373767b7e06508eb48 100644
--- a/dist/index.es.js
+++ b/dist/index.es.js
@@ -8256,7 +8256,7 @@ class CubismRenderer_WebGL extends CubismRenderer {
   release() {
     var _a, _b, _c;
     const self2 = this;
-    this._clippingManager.release();
+    this._clippingManager?.release();
     self2._clippingManager = void 0;
     (_a = this.gl) == null ? void 0 : _a.deleteBuffer(this._bufferData.vertex);
     this._bufferData.vertex = null;
diff --git a/dist/index.js b/dist/index.js
index 3f7a031f8602cf0566e2d7ea4e2b739419f57c81..b659f85e420f6179112f59cf238dd66f322aaba3 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -8258,7 +8258,7 @@ var __async = (__this, __arguments, generator) => {
     release() {
       var _a, _b, _c;
       const self2 = this;
-      this._clippingManager.release();
+      this._clippingManager?.release();
       self2._clippingManager = void 0;
       (_a = this.gl) == null ? void 0 : _a.deleteBuffer(this._bufferData.vertex);
       this._bufferData.vertex = null;
